File: Pieces/Piece.java
package chessfinalproject.Pieces;

import java.util.ArrayList;

public class Piece
	{
	String position; // Stores position in algebraic chess notation (a1-h8)
	int piece_ident; // Stores the type of piece this is according to the
						// following numeric code:
	/*
	 * 0  = No Piece
	 * 1  = White King
	 * 2  = White Queen
	 * 3  = White Bishop
	 * 4  = White Knight
	 * 5  = White Rook
	 * 6  = White Pawn
	 * 7  = Black King
	 * 8  = Black Queen
	 * 9  = Black Bishop
	 * 10 = Black Knight
	 * 11 = Black Rook
	 * 12 = Black Pawn
	 */
	
	public Piece()
		{
		piece_ident = 0;
		}
	
	public String[] get_moves() // Returns an array containing the possible
								// moves for the piece.
		{
		String[] default_return = {"null"};
		
		return default_return;
		}
	
	}

File: chess/piece.py
'''
Piece class for Computer Science chess final project

Contains all methods related to piece movement 
'''

class Piece:
	piece_type, color, position = "Null"
	
	def __init__(self, piece_type, color, position): 
		self.piece_type = piece_type
		self.color = color
		self.position = position
		''' piece_type is a string that describes which piece it is.
			color refers to which player the piece belongs (black or white)
			position describes the location of the piece on the board (e.g. a5)'''
		
	def get_moves(self):
		'''
		if self.piece_type == "pawn":
			
		elif self.piece_type == "rook":
			#TODO: ...
		elif self.piece_type == "knight":
			#TODO: ...
		elif self.piece_type == "bishop":
			#TODO: ...
		elif self.piece_type == "queen":
			#TODO: ...
		elif self.piece_type == "king":
			#TODO: ...
		else:
			return "Error: No piece type defined for object #" + id(self)
			'''
		
		print("TODO: ...")
